src/config/mongoose_config_parser_toml.erl

```diff
@@ -1365,21 +1365,44 @@ process_acl([ACLName, _|Path], Content) ->
     [acl:to_record(host(Path), b2a(ACLName), acl_data(Content))].
 
 -spec acl_data(toml_value()) -> option().
-acl_data(Content) when is_map(Content) ->
-    case maps:to_list(Content) of
-        [{Key, Values}] when is_list(Values) ->
-            list_to_tuple([b2a(Key) | Values]);
-        [{Key, Value}] when is_binary(Value) ->
-            {b2a(Key), Value}
-    end;
-acl_data(Value) when is_binary(Value) -> b2a(Value).
+acl_data(<<"all">>) -> all;
+acl_data(<<"none">>) -> none;
+acl_data(M) ->
+    {AclName, AclKeys} = find_acl(M, lists:sort(maps:keys(M)), acl_keys()),
+    list_to_tuple([AclName | lists:map(fun(K) -> maps:get(K, M) end, AclKeys)]).
```

Comment:
If I understand this line correctly, I would find something like this more readable, but it's a matter of taste.
```suggestion
    list_to_tuple([AclName | maps:to_list(maps:with(AclKeys, M))).
```

Reply:
Not quite the same, your version changes the ordering of the keys.